### ISSUE TITLE
ninja jobs explode when nesting.

### ISSUE THREAD

**Issue by graph** [ninja-build/ninja#1253]
The default on my machine is 6 jobs. I have nesting going on, and don't have control over all the nested scripts. I was looking for a environment variable to change the default to 1 but all those issues have been closed. Just 2 nests thats 6*6, I have 3 nests that's 6*6*6 processes doing high intensity work. This makes my system unusable and slows down overall the entire build.

One idea is to use file locking or inteprocess communication to sync total number of ninja jobs and have a minimum of 1 job per nest to prevent deadlocking.

Just a simple environment variable works too.

Thank you.

**Comment by evmar** [comment]
This would be addressed by https://github.com/ninja-build/ninja/issues/1139

**Comment by graph** [comment]
Yeah it looks like it would. That issue has been open for a while. if doing the jobserver approach will take much time I think it would be best to introduce a environment variable for the short term.

**Comment by tcmal** [comment]
#1139 adds a jobserver client, but since there is no *server* implementation in ninja, it seems that nesting will still blow past `-j` limits unless you have a top-level `make` instance above ninja.

We're using this hack for now:

```
echo -e "it:\n\tninja $TARGET ${EXTRA_ARGS[@]}" > Makefile
make -j$NPROC it
```

It would be good if ninja implemented the server part of the jobserver. Failing that, the fact that nesting doesn't 'just work' should be documented.

**Comment by thesamesam** [comment]
See https://github.com/ninja-build/ninja/pull/2634 for the server impl.
